utils/http.go

```diff
@@ -28,74 +29,137 @@ type rootHandler struct {
 	context context.Context
 	trust   signed.CryptoService
 	//cachePool redis.Pool
+	imageNameAt ImageNameLocation
+}
+
+// ImageNameLocation is used to define where the auth code should read the image name
+// from. For most repo operations, it will be ImageInURL, i.e. it's the segments between
+// `/v2/` and `/_trust`. For the changefeed however, an image name is optional, and if
+// present will be found in the query string.
+type ImageNameLocation int
+
+// Constants for different image locations
+const (
+	ImageInURL = iota
+	ImageInQueryString
+	NoImageName
+)
+
+func parseImageName(loc ImageNameLocation, r *http.Request) (string, error) {
+	switch loc {
+	case ImageInURL:
+		vars := mux.Vars(r)
+		return vars["imageName"], nil
+	case ImageInQueryString:
+		qs := r.URL.Query()
+		imageName := qs.Get("filter")
+		if imageName == "" {
+			// no image name means global feed
+			imageName = "*"
+		}
+		return imageName, nil
+	case NoImageName:
```

Comment:
This doesn't indicate no image was passed, it indicates no image was expected. The `imageName == ""` above is to set a default value when we expected an image name and none was explicitly given (and in the changefeed situation I put that in for, it's desirable to make that assumption when no name is given).

I opened issue #1021 to look at refactoring how we do auth in general because this is only one of the things I wasn't very happy with, regardless of how it's named.